File: Java/DesignPatterns/src/test/java/designPatterns/behavioral/visitor/MainVisitorTest.java
package designPatterns.behavioral.visitor;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainVisitorTest {
  private Part part;

  @Test
  void equalsTest() {
    part = new ConcretePart.A();
    assertThat(part.accept(new DisplayVisitor())).isEqualTo("Displaying A");

    part = new ConcretePart.B();
    assertThat(part.accept(new DisplayVisitor())).isEqualTo("Displaying B");

    part = new ConcretePart.C();
    assertThat(part.accept(new DisplayVisitor())).isEqualTo("Displaying C");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/behavioral/visitor/MainVisitorTest.kt
package behavioral.visitor

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class MainVisitorTest {
  private lateinit var part: Part

  @Test
  internal fun equalsTest() {
    part = ConcretePart.A()
    assertThat(part.accept(DisplayVisitor())).isEqualTo("Displaying A")

    part = ConcretePart.B()
    assertThat(part.accept(DisplayVisitor())).isEqualTo("Displaying B")

    part = ConcretePart.C()
    assertThat(part.accept(DisplayVisitor())).isEqualTo("Displaying C")
  }
}